File: frontend/src/types/thesis.ts
export interface EvidenceStep {
  step: number;
  description: string;
  source_company: string;
  source_filing: string;
  quote: string;
}

export type ThesisType =
  | "supply_chain_contagion"
  | "sector_read_through"
  | "strategic_pivot"
  | "peer_comparison";

export type ThesisDirection = "long" | "short" | "long_short_pair";

export type ThesisStatus = "pending" | "validated" | "dismissed" | "expired";

export interface Thesis {
  id: string;
  title: string;
  summary: string;
  thesis_type: ThesisType;
  direction: ThesisDirection;
  confidence_score: number;
  trigger_company_id: string;
  affected_company_ids: string[];
  trigger_ticker: string | null;
  affected_tickers: string[];
  evidence_chain: EvidenceStep[];
  competing_thesis: string | null;
  invalidation_criteria: string[];
  catalyst: string | null;
  time_horizon: string | null;
  status: ThesisStatus;
  pm_notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface ThesisListResponse {
  theses: Thesis[];
  total: number;
  limit: number;
  offset: number;
}

export interface ThesisStats {
  total: number;
  pending: number;
  validated: number;
  dismissed: number;
  expired: number;
}

export interface ThesisFilters {
  status?: ThesisStatus;
  direction?: ThesisDirection;
  confidence_min?: number;
  limit?: number;
  offset?: number;
}

File: backend/app/schemas/thesis.py
"""Pydantic schemas for Theses — the user-facing output of the pipeline."""
from __future__ import annotations

import uuid
from datetime import datetime
from enum import Enum

from pydantic import BaseModel, ConfigDict, Field


class ThesisType(str, Enum):
    SUPPLY_CHAIN_CONTAGION = "supply_chain_contagion"
    SECTOR_READ_THROUGH = "sector_read_through"
    STRATEGIC_PIVOT = "strategic_pivot"
    PEER_COMPARISON = "peer_comparison"


class ThesisDirection(str, Enum):
    LONG = "long"
    SHORT = "short"
    LONG_SHORT_PAIR = "long_short_pair"


class ThesisStatus(str, Enum):
    PENDING = "pending"
    VALIDATED = "validated"
    DISMISSED = "dismissed"
    EXPIRED = "expired"


class EvidenceStep(BaseModel):
    step: int = Field(..., ge=1)
    description: str
    source_company: str
    source_filing: str
    quote: str


class ThesisCreate(BaseModel):
    title: str = Field(..., max_length=500)
    summary: str
    thesis_type: ThesisType
    direction: ThesisDirection
    confidence_score: float = Field(..., ge=0.0, le=1.0)

    trigger_company_id: uuid.UUID
    affected_company_ids: list[uuid.UUID] = Field(default_factory=list)

    evidence_chain: list[EvidenceStep] = Field(default_factory=list)
    competing_thesis: str | None = None
    invalidation_criteria: list[str] = Field(default_factory=list)

    catalyst: str | None = None
    time_horizon: str | None = None

    trigger_delta_ids: list[uuid.UUID] = Field(default_factory=list)
    trigger_fact_ids: list[uuid.UUID] = Field(default_factory=list)


class ThesisUpdate(BaseModel):
    status: ThesisStatus | None = None
    pm_notes: str | None = None


class ThesisResponse(BaseModel):
    id: uuid.UUID
    title: str
    summary: str
    thesis_type: ThesisType
    direction: ThesisDirection
    confidence_score: float

    trigger_company_id: uuid.UUID
    affected_company_ids: list[uuid.UUID]
    trigger_ticker: str | None = None
    affected_tickers: list[str] = Field(default_factory=list)

    evidence_chain: list[EvidenceStep]
    competing_thesis: str | None
    invalidation_criteria: list[str]

    catalyst: str | None
    time_horizon: str | None

    status: ThesisStatus
    pm_notes: str | None = None

    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ThesisListResponse(BaseModel):
    theses: list[ThesisResponse]
    total: int
    limit: int
    offset: int


class ThesisStatsResponse(BaseModel):
    total: int
    pending: int
    validated: int
    dismissed: int
    expired: int


class ValidateRequest(BaseModel):
    notes: str | None = None


class DismissRequest(BaseModel):
    reason: str | None = None
